src/java.base/share/classes/java/text/MessageFormat.java

```diff
@@ -390,11 +390,21 @@ public MessageFormat(String pattern) {
      * Patterns and their interpretation are specified in the
      * <a href="#patterns">class description</a>.
      *
+     * @implSpec The default implementation throws a
+     * {@code NullPointerException} if {@code locale} is {@code null}
+     * either during the creation of the {@code MessageFormat} object or later
+     * when {@code format()} is called by the constructed {@code MessageFormat}
+     * instance and the implementation utilizes a subformat that requires
+     * localization.
```

Comment:
Sorry for the nitpick, but since localization itself is not a requirement, `locale-dependent subformat` would read better to me. (and wherever `localized` is used)